Fix for SNAP-1461

kneeraj opened the pull request:
## Changes proposed in this pull request

In fact any query with more than one table will be routed to snappy spark now and not only the problematic query mentioned in the issue SNAP-1461

## Patch testing

dunit

## ReleaseNotes changes

None.

## Other PRs 

None.

Commit:
Fix for SNAP-1461

sumwale commented on gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/rules/AnyOneOfExecutionEngineRule.java:
```diff
@@ -59,6 +62,11 @@ protected ExecutionEngine findExecutionEngine(DMLQueryInfo qInfo, ExecutionRuleC
       }
     }
+
+    List<TableQueryInfo> tqis = qInfo.getTableQueryInfoList();
```
Add a comment here what this means. This class can use a general javadoc stating the rules being applied here for routing.

sumwale reviewed (approved):
LGTM. See comment about javadoc.

Commit:
Changed the sequence of the routing rule to apply set of ANy rules before replicated table rules.

Commit:
Merge branch 'snappy/master' into SNAP-1461

Commit:
Made the routing proper now. Replicated tables expensive queries will also be routed.
Added a state in the execution context for identifying system tables.

Commit:
Cannot blindly change the rule execution order unless more changes can be done
as sys table queries are on replicated tables and they need to be filtered out.
So keeping the order same and then applying the anyoneof rule.

kneeraj commented:
Instead of changing the order of applying of rules to push the replicated table rules below now I have kept the order same mainly to filter the sys table queries before considerign the anyone of rules for replicated table too.
